# Run stylelint on inline CSS

## Rule Details

This rule checks that inline CSS is valid and respects Primer's [stylelint config](https://github.com/primer/stylelint-config).

## Examples
### **Incorrect** code for this rule ğŸ‘

```erb
<!-- incorrect -->
<div style="color: black"></div>
```

### **Correct** code for this rule  ğŸ‘

```erb
<!-- correct -->
<div style="color: var(--fgColor-default)"></div>
```
